GliaWidgets/Sources/CallVisualizer/VideoCall/ViewModel/VideoCallViewModel.swift

```diff
@@ -0,0 +1,590 @@
+        return .init(
+            image: operatorImage?.image,
+            animated: operatorImage?.animated ?? false
+        )
+    }
+
+    private func makeUserImageProps() -> VideoCall.UserImageView.Props {
+        let operatorImageProps = makeOperatorImageViewProps()
+        return .init(
+            style: style.connect.connectOperator.operatorImage,
+            operatorImageViewProps: operatorImageProps,
+            operatorImageVisible: operatorImageVisible,
+            placeHolderImage: operatorImagePlaceholder
+        )
+    }
+
+    private func makeConnetOperatorViewProps() -> VideoCall.ConnectOperatorView.Props {
+        let userImageProps = makeUserImageProps()
+        return .init(
+            style: style.connect.connectOperator,
+            size: .init(
+                size: connectOperatorSize.size,
+                animated: connectOperatorSize.animated
+            ),
+            operatorAnimate: animateOperator,
+            userImageViewProps: userImageProps
+        )
+    }
+
+    private func makeConnectStatusViewProps() -> VideoCall.ConnectStatusView.Props {
+        return .init(
+            style: statusStyle,
+            firstLabelText: .init(
+                text: statusFirstText.text,
+                animated: statusFirstText.animated
+            ),
+            secondLabelText: .init(
+                text: statusSecondText.text,
+                animated: statusSecondText.animated
+            )
+        )
+    }
+
+    private func makeConnectViewProps() -> VideoCall.ConnectView.Props {
+        let connectOperatorViewProps = makeConnetOperatorViewProps()
+        let connectStatusViewProps = makeConnectStatusViewProps()
+        let connectViewProps: VideoCall.ConnectView.Props = .init(
+            operatorViewProps: connectOperatorViewProps,
+            statusViewProps: connectStatusViewProps,
+            state: state,
+            transition: connectVisibility
+        )
+
+        return connectViewProps
+    }
+
+    private func makeButtonBarProps() -> VideoCall.CallButtonBar.Props {
+        return .init(
+            style: style.buttonBar,
+            videoButtonTap: .init { [weak self] in
+                self?.toggleVideo()
+            },
+            minimizeTap: .init { [weak self] in
+                self?.delegate?(.minimized)
+            },
+            videoButtonState: videoButtonState,
+            videoButtonEnabled: videoButtonEnabled,
+            minimizeButtonEnabled: minimizeButtonEnabled,
+            effect: interfaceOrientation.isLandscape ? .blur : .none
+        )
+    }
+}
+
+// MARK: - ConnectView Handler
+
+extension CallVisualizer.VideoCallViewModel {
+    func setConnectViewState(_ state: State, animated: Bool) {
+        self.state = state
+        switch state {
+        case .initial:
+            break
+        case .queue:
+            break
+        case .connecting(let name, let imageUrl):
+            animateOperator = animated
```

Comment:
looks like value changes on almost every line will trigger `reportUpdate()`

Reply:
Maybe @igorkravchenko can share some insight here. But as much as I know, this is the only drawback at the moment with this method. Good thing, most of them won't be applied since we check for the old value and so we won't rerender all the views

Reply:
Correct, if state is updated on view model, props that depend on that state have to be regenerated. Some of these  update are handled by views, rendering only in case if there is difference from old value.

Reply:
What do you think about adding something like debounce wrapper for `reportUpdate()` with 300 msec or even less to decrease amount of View layer updates?
@igorkravchenko @rasmustautsglia

Reply:
Reopened thread to discuss point above

Reply:
That would work in normal cases, however since video and audio stream are part of the props, I believe this will cause unexpected behavior